connectors/handlers/views.py

```diff
@@ -111,3 +116,25 @@ def slack_bot_handle_callback_events(request_message: HttpRequest) -> JsonRespon
         else:
             return JsonResponse({'success': False, 'message': f"Received invalid data type: {d_type}"}, status=400)
     return JsonResponse({'success': False, 'message': 'Slack Event Callback  Handling failed'}, status=400)
+
+
+@csrf_exempt
+@api_view(['POST'])
+def pagerduty_handle_incidents(request_message: HttpRequest) -> JsonResponse:
+    try:
+        data = request_message.data
+        handle_pd_incident(data)
+        return JsonResponse({'success': False, 'message': 'pagerduty incident Handling failed'}, status=200)
+    except Exception as e:
+        logger.error(f'Error handling pagerduty incident: {str(e)}')
+        return JsonResponse({'success': False, 'message': f"pagerduty incident Handling failed"}, status=500)
+
+
+@web_api(ExecuteWorkflowRequest)
+def pd_generate_webhook(request_message: ExecuteWorkflowRequest) -> HttpResponse:
+    location = '/connectors/handlers/pagerduty/handle_incidents/'
+    protocol = settings.WORKFLOW_EXECUTE_API_SITE_HTTP_PROTOCOL
```

Comment:
Create new varianle in base_settings.py : 

PAGERDUTY_WEBHOOK_HTTP_PROTOCOL
and
PAGERDUTY_WEBHOOK_USE_SITE